packages/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx

```diff
@@ -0,0 +1,615 @@
+  }
+
+  private _getMargins = (margins: IMargins) => {
+    this.margins = margins;
+  };
+
+  private _renderContentForOnlyBars = (props: IHorizontalBarChartWithAxisDataPoint): JSX.Element => {
+    const { useSingleColor = false } = this.props;
+    return (
+      <>
+        <ChartHoverCard
+          XValue={props.xAxisCalloutData || props.x.toString()}
+          Legend={props.legend}
+          YValue={props.yAxisCalloutData || props.y}
+          color={!useSingleColor && props.color ? props.color : this._createColors()(props.x)}
+          culture={this.props.culture}
+        />
+      </>
+    );
+  };
+
+  private _renderCallout = (props?: IHorizontalBarChartWithAxisDataPoint): JSX.Element | null => {
+    return props ? this._renderContentForOnlyBars(props) : null;
+  };
+
+  private _getCustomizedCallout = () => {
+    return this.props.onRenderCalloutPerDataPoint
+      ? this.props.onRenderCalloutPerDataPoint(this.state.dataPointCalloutProps, this._renderCallout)
+      : null;
+  };
+
+  private _getGraphData = (
+    xScale: StringAxis,
+    yScale: NumericAxis,
+    containerHeight: number,
+    containerWidth: number,
+    xElement?: SVGElement | null,
+    yElement?: SVGElement | null,
+  ) => {
+    return (this._bars =
+      this._yAxisType === YAxisType.NumericAxis
+        ? this._createNumericBars(containerHeight, containerWidth, xElement!, yElement!)
+        : this._createStringBars(containerHeight, containerWidth, xElement!, yElement!));
+  };
+
+  private _createColors(): D3ScaleLinear<string, string> | ColorScale {
+    const increment = this._colors.length <= 1 ? 1 : 1 / (this._colors.length - 1);
+    const { useSingleColor = false } = this.props;
+    if (useSingleColor) {
+      return (_p?: number) => {
+        const { theme, colors } = this.props;
+        return colors && colors.length > 0 ? colors[0] : theme!.palette.blueLight;
+      };
+    }
+    const domainValues = [];
+    for (let i = 0; i < this._colors.length; i++) {
+      domainValues.push(increment * i * this._xMax);
+    }
+    const colorScale = d3ScaleLinear<string>().domain(domainValues).range(this._colors);
+    return colorScale;
+  }
+
+  private _refCallback = (element: SVGRectElement, legendTitle: string): void => {
+    this._refArray.push({ index: legendTitle, refElement: element });
+  };
+
+  private _getCalloutContentForLineAndBar = (
+    point: IHorizontalBarChartWithAxisDataPoint,
+  ): { YValueHover: IYValueHover[]; hoverXValue: string | number | null } => {
+    const YValueHover: IYValueHover[] = [];
+    const { useSingleColor = false } = this.props;
+    const { data } = this.props;
+    const selectedPoint = data!.filter((xDataPoint: IHorizontalBarChartWithAxisDataPoint) => xDataPoint.x === point.x);
+    // callout data for the bar
+    YValueHover.push({
+      legend: selectedPoint[0].legend,
+      y: selectedPoint[0].x,
```

Comment:
x is assigned to y. Do we need to change the name of structure to represent this swap